build/tasks/build.js

```diff
@@ -31,45 +32,93 @@ gulp.task('build-index', function(){
     .pipe(gulp.dest(paths.output));
 });
 
-gulp.task('build-es2015', function () {
+var indexSrc = gulp
+  .src(paths.output + paths.packageName + '.js')
+  .pipe(rename(function (path) {
+    if (path.extname == '.js') {
+      path.extname = '.ts'
+    }
+  }))
+
+gulp.task('build-ts-es2015', function () {
+  var tsProjectES2015 = ts.createProject(compilerTsOptions.es2015(), ts.reporter.defaultReporter());
+  var tsResult = indexSrc.pipe(ts(tsProjectES2015));
+  return tsResult.js
+    .pipe(gulp.dest(paths.output + 'es2015'));
+});
+
+gulp.task('build-ts-commonjs', function () {
+  var tsProjectCommonJS = ts.createProject(compilerTsOptions.commonjs(), ts.reporter.defaultReporter());
+  var tsResult = indexSrc.pipe(ts(tsProjectCommonJS));
+  return tsResult.js
+    .pipe(gulp.dest(paths.output + 'commonjs'));
+});
+
+gulp.task('build-ts-amd', function () {
+  var tsProjectAmd = ts.createProject(compilerTsOptions.amd(), ts.reporter.defaultReporter());
+  var tsResult = indexSrc.pipe(ts(tsProjectAmd));
+  return tsResult.js
+    .pipe(gulp.dest(paths.output + 'amd'));
+});
+
+gulp.task('build-ts-system', function () {
+  var tsProjectSystem = ts.createProject(compilerTsOptions.system(), ts.reporter.defaultReporter());
+  var tsResult = indexSrc.pipe(ts(tsProjectSystem));
+  return tsResult.js
+    .pipe(gulp.dest(paths.output + 'system'));
+});
+
```

Comment:
Content of all `build-ts-*` tasks could be easily refactored to single function that takes module format as an argument and that could be internally passed to `compilerTsOptions.compilerTsOptionsForModule(module)` (see my other comment regarding compilerTsOptionsForModule)